src/pyodide/internal/pyodide-bootstrap.js

```diff
@@ -175,15 +158,30 @@ export default {
 
       // Apply patches that enable some packages to work.
       const patches = getPatches();
-      // TODO(EW-8055): Why does micropip.list not work?
-      if (JSON.parse(micropip.freeze())["packages"]["aiohttp"] !== undefined) {
+      if (micropip.list().has("aiohttp") !== undefined) {
         pyodide.runPython(patches["aiohttp_fetch_patch.py"]);
       }
     }
 
-
     await pyodide.loadPackage(pythonRequirements);
 
     return await pyodide.pyimport(metadata.mainModule).fetch(request);
   },
+  async test() {
+    try {
+      const metadata = transformMetadata(getMetadata());
+      const pyodide = await loadPyodide();
+      for (const { name, value } of metadata.globals) {
+        if (value.pythonModule !== undefined) {
+          pyodide.FS.writeFile(`/session/${name}.py`, value.pythonModule, {
+            canOwn: true,
+          });
+        }
+      }
+      await pyodide.pyimport(metadata.mainModule).test();
+    } catch (e) {
+      console.log(e);
+      console.log(e.stack);
+    }
+  },
```

Comment:
I guess this is just testing code that can be removed.

```suggestion

```

Reply:
I think it's kind of useful to have something like this to handle a `test` entrypoint (I've certainly been using it a lot). But I'll  remove this here and add something more carefully considered in #1566.